framework/src/wifi_manager/wifi_manager.c

```diff
@@ -635,6 +647,7 @@ wifi_manager_result_e _get_ipaddr_dhcp(void)
 		if (ret != OK) {
 			ndbg("[WM] get IP address fail\n");
 			dhcpc_close(dhcp_hnd);
+			WIFIADD_ERR_LOG(ERR_WIFIMGR_CONNECT_FAIL);
```

Comment:
Because WiFi Manager knows dhcp failure here, why don't you make another filed, e.g., ERR_WIFIMGR_(CONNECT_)DHCP_FAIL, as long as you do not have a plan to extend the err log system into dhcp code.  It might the same logical structure that applies to MSGQ such as ERR_WIFIMGR_MSGQ_FAIL. 
You may consider start/stop_dhcpd in SoftAP mode, too.

Reply:
Thanks for the comment. I agree with the general tone here, that we will benefit from a more specific reason for failure (DHCP failure). Ideally though, the error code for dhcp failure shoould be captured in the dhcpc module (and likewise, dhcpd for server mode). The downside to modularity though, is the increased overhead in maintaining error reporting across many such modules. We need to have a clear policy on how deep we report errors at a given module. As for MSGQ IPC, it falls outside the scope of network, so we can get specific at the level of WiFi Manager. This was the reasoning during development, please let me have your suggestions.

Reply:
DHCP is a part of the network, therefore I totally agree with you to make it have its own error reporting module. However, because WiFi Manager directly calls DHCP functionalities and knows its success and failure, I think it might be useful to include error reporting of DHCP in WiFi Manager, which may help to reduce the overhead potentially. Anyway, as you mentioned, we need to decide a clear policy, considering the purpose of the system. Thank you.

Reply:
Thanks for your comments. Yes, DHCP's status can somewhat be read as an aggregate at the level of WiFi Manager. I will make the changes you suggested.